File: src/main/java/Application/Services/ExportReportService.java
package Application.Services;

import Application.Features.Reports.Documentary.GetDocumentaryReportResponse;
import Application.Features.Reports.Financial.GetFinancialReportResponse;
import Application.Features.Reports.Operational.GetOperationalReportResponse;

public interface ExportReportService {
    byte[] exportFinancial(GetFinancialReportResponse report, String format) throws Exception;
    byte[] exportDocumentary(GetDocumentaryReportResponse report, String format) throws Exception;
    byte[] exportOperational(GetOperationalReportResponse report, String format) throws Exception;
}

File: src/main/java/Application/Features/Reports/Export/ExportReportService.kt
package Application.Services

import Application.Features.Reports.Documentary.GetDocumentaryReportResponse
import Application.Features.Reports.Financial.GetFinancialReportResponse
import Application.Features.Reports.Operational.GetOperationalReportResponse

interface ReportExportService {
    @Throws(Exception::class)
    fun exportFinancial(report: GetFinancialReportResponse?, format: String?): ByteArray?

    @Throws(Exception::class)
    fun exportDocumentary(report: GetDocumentaryReportResponse?, format: String?): ByteArray?

    @Throws(Exception::class)
    fun exportOperational(report: GetOperationalReportResponse?, format: String?): ByteArray?
}